### PR TITLE
Create structural metadata using cocina

### PR DESCRIPTION
**Pull request by jcoyne**
Ref #654

## How was this change tested? 🤨

⚡ ⚠ If this change involves consuming from other services or writing to shared file systems, test that web archive seed and crawl accessioning works properly in [stage|qa] environment, in addition to specs. ⚡

### CODE REVIEW

File: lib/dor/was_seed/structural_builder.rb [hunk 1]
```diff
@@ -0,0 +1,103 @@
+# frozen_string_literal: true
+
+module Dor
+  module WasSeed
+    # Builds a Cocina representation of the structural metadata
+    class StructuralBuilder
+      # Generates structural metadata for a repository object.
+      #
+      # @param [Cocina::Models::DRO] cocina_model
+      # @param [Array<Array<Assembly::ObjectFile>>] objects the list of files to add to structural metadata
```
Comment (justinlittman) [inline lib/dor/was_seed/structural_builder.rb:10]: Isn't there always only a single ObjectFile? Seems like this has additional complexity to support an array.

File: lib/dor/was_seed/structural_builder.rb [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+      # @param [String] reading_order ('left-to-right') only valid for books, can be 'right-to-left' or 'left-to-right'.
```
Comment (justinlittman) [inline lib/dor/was_seed/structural_builder.rb:11]: reading_order is cruft?

File: lib/dor/was_seed/structural_builder.rb [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+      def self.build(cocina_model:, objects:, reading_order: 'left-to-right')
+        common_path = find_common_path(objects) # find common paths to all files provided
+
+        filesets = objects.map { |resource_files| FileSet.new(resource_files:, cocina_model:) }
+
+        structural = {
+          contains: build_filesets(filesets:, cocina_model:, common_path:)
+        }
+
+        structural[:hasMemberOrders] = [{ viewingDirection: reading_order }] if cocina_model.type == Cocina::Models::ObjectType.book
```
Comment (justinlittman) [inline lib/dor/was_seed/structural_builder.rb:21]: cruft?

File: lib/dor/was_seed/structural_builder.rb [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+          raise "File '#{obj.path}' not found" unless obj.file_exists?
+
+          obj.path # collect all of the filenames into an array
+        end
+
+        ::Assembly::ObjectFile.common_path(all_paths) # find common paths to all files provided if needed
+      end
+      private_class_method :find_common_path
+
+      def self.administrative(file_attributes, object_access)
```
Comment (justinlittman) [inline lib/dor/was_seed/structural_builder.rb:36]: Can't administrative just be hardcoded for the single file?

File: lib/dor/was_seed/structural_builder.rb [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+      end
+      private_class_method :administrative
+
+      def self.build_filesets(filesets:, cocina_model:, common_path:)
+        # a counter to use when creating auto-labels for resources, with incremenets for each type
+        resource_type_counters = Hash.new(0)
+
+        pid = cocina_model.externalIdentifier.delete_prefix('druid:') # remove druid prefix when creating IDs
+
+        filesets.map.with_index(1) do |fileset, sequence| # iterate over all the resources
```
Comment (justinlittman) [inline lib/dor/was_seed/structural_builder.rb:51]: There's only ever one fileset.

**Review by justinlittman — CHANGES_REQUESTED** [review]

**Comment by justinlittman** [comment]
This seems excessively complex for the structural for a single file.

**Comment by jcoyne** [comment]
This was copied from common-accessioning. We can make it simpler, but this approach is tried and tested.

**Comment by justinlittman** [comment]
In GIS Robot Suite it is only as complicated as it needs to be: https://github.com/sul-dlss/gis-robot-suite/blob/main/lib/robots/dor_repo/gis_assembly/generate_structural.rb

**Review by justinlittman — APPROVED** [review]

**Comment by justinlittman** [comment]
Holding for testing by @andrewjbtw

**Pull request merged**